lib/index.js

```diff
@@ -179,6 +200,39 @@ FSTree.defaultIsEqual = function defaultIsEqual(entryA, entryB) {
   return equal;
 };
 
+FSTree.applyPatch = function(input, output, patch, _delegate) {
+  var delegate = assign({}, DEFAULT_DELEGATE, _delegate);
+  for (var i = 0; i < patch.length; i++) {
+    applyOperation(input, output, patch[i], delegate);
+  }
+};
+
+function applyOperation(input, output, operation, delegate) {
+  var method = operation[0];
+  var relativePath = operation[1];
+  var inputPath = path.join(input, relativePath);
+  var outputPath = path.join(output, relativePath);
+
+  if (delegate[method]) {
+    delegate[method](inputPath, outputPath, relativePath);
+  } else {
+    throw new Error('Unknown patch operation method: ' + method);
+  }
+}
+
+function copy(sourcePath, outputPath) {
+  var outputDir = path.dirname(outputPath);
+  try {
+    symlinkOrCopy.sync(sourcePath, outputPath);
+  } catch (e) {
+    try {
```

Comment:
i will share why, so it can be added:

* we are gambling that the directory already exists, if it does not we make the directory then apply the symlink. In-practice apps tend to have many files, and low number of directories. So this mitigates the overhead of defensively mkdirp.sync'ing always.